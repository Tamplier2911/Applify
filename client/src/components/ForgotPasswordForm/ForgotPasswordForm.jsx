// import "./ForgotPasswordForm.scss";
import React, { useState } from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";
import { openModal } from "../../redux/modal/modal.actions";
import { userForgotPasswordStart } from "../../redux/auth/auth.actions";

// validator
import emailValidator from "../../utils/EmailValidator/emailValidator";

// components
import FormInput from "../FormInput/FormInput";
import Button from "../Button/Button";

// JS Rendering CSS
import {
  ForgotPasswordFormContainer,
  ForgotPasswordFormTitle,
  ForgotPasswordFormElement
} from "./ForgotPasswordFormStyles";

// component constants
import forgotPasswordFormData from "../../utils/ComponentForgotPasswordFormConstants/componentForgotPasswordFormConstants";

const ForgotPasswordForm = ({ lang, openModal, userForgotPasswordStart }) => {
  const [userEmailAddress, setUserEmailAddress] = useState({ userEmail: "" });
  const { userEmail } = userEmailAddress;

  const onInputChange = e => {
    const { name, value } = e.target;
    setUserEmailAddress({ ...userEmailAddress, [name]: value });
  };

  const onSubmit = e => {
    e.preventDefault();
    if (!emailValidator(userEmail) || !userEmail)
      return openModal({
        header: "Attention!",
        content: "Please, enter your email address."
      });
    userForgotPasswordStart(userEmailAddress);
  };

  const {
    forgotPasswordFormTitle,
    forgotPasswordEmail,
    forgotPasswordFormButton
  } = forgotPasswordFormData[lang];

  return (
    <ForgotPasswordFormContainer>
      <ForgotPasswordFormTitle lang={lang}>
        {forgotPasswordFormTitle}
      </ForgotPasswordFormTitle>
      <ForgotPasswordFormElement onSubmit={e => onSubmit(e)} autoComplete="on">
        <FormInput
          onInputChange={e => onInputChange(e)}
          value={userEmail}
          label={forgotPasswordEmail}
          name="userEmail"
          type="email"
          required
        />
        <Button type="submit" value={forgotPasswordFormButton} />
      </ForgotPasswordFormElement>
    </ForgotPasswordFormContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage
});

export default connect(mapStateToProps, { openModal, userForgotPasswordStart })(
  ForgotPasswordForm
);
