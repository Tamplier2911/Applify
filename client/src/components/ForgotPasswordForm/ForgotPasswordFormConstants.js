const forgotPasswordFormData = {
  eng: {
    forgotPasswordFormTitle:
      "Please, enter your email address, and we will send you instructions on password restoration process.",
    forgotPasswordEmail: "Email",
    forgotPasswordFormButton: "Send",
  },
  rus: {
    forgotPasswordFormTitle:
      "Пожалуйста, введите свой адрес электронной почты, и мы вышлем вам инструкции по восстановлению пароля.",
    forgotPasswordEmail: "Эл. адрес",
    forgotPasswordFormButton: "Отправить",
  },
  ukr: {
    forgotPasswordFormTitle:
      "Будь ласка, введіть свою адресу електронної пошти, і ми надішлемо вам інструкції з відновлення пароля.",
    forgotPasswordEmail: "Ел. адреса",
    forgotPasswordFormButton: "Відправити",
  },
};

export default forgotPasswordFormData;
