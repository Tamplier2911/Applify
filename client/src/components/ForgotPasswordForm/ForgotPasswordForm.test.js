import React from "react";
import toJson from "enzyme-to-json";

// component
import ForgotPasswordForm from "./ForgotPasswordForm";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<ForgotPasswordForm />", () => {
  const initialStore = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<ForgotPasswordForm store={store} />);
  });

  it("renders without an error", () => {
    const forgotPasswordForm = findByTestAttr(wrapper, "forgot-password-form");
    expect(forgotPasswordForm.length).toBe(1);
  });

  it("renders children - forgot password form element", () => {
    const forgotPasswordFormElement = findByTestAttr(
      wrapper,
      "forgot-password-form-element"
    );
    expect(forgotPasswordFormElement.length).toBe(1);
  });

  it("matches snapshot", () => {
    const forgotPasswordForm = findByTestAttr(wrapper, "forgot-password-form");
    expect(toJson(forgotPasswordForm)).toMatchSnapshot();
  });
});
