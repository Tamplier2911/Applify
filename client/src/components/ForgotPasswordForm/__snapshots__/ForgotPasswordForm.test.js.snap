// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPasswordForm /> matches snapshot 1`] = `
<styled.div
  data-test="forgot-password-form"
>
  <styled.h3
    lang="eng"
  >
    Please, enter your email address, and we will send you instructions on password restoration process.
  </styled.h3>
  <styled.form
    autoComplete="on"
    data-test="forgot-password-form-element"
    onSubmit={[Function]}
  >
    <FormInput
      label="Email"
      name="userEmail"
      onInputChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <Button
      type="submit"
      value="Send"
    />
  </styled.form>
</styled.div>
`;
